README.md

```diff
@@ -496,63 +491,86 @@ The `auth` field MUST be contain an [Authorization] which signs over the `&Task`
 
 Concretely, this means that the `&Task` MUST be present in the associated `auth`'s `scope` field. An `Invocation` where the associated [Authorization] does not include the [Task] in the `scope` MUST be considered invalid.
 
-## 5.1 Schema
+## 5.1 Context
 
 ```ipldsch
-type Invocation struct {
-  v       SemVer
-
+type Context struct {
   run     &Task
+  meta    {String : Any}
+  prf     [&UCAN]
+
   # Receipt of the invocation that caused this invocation
   cause   optional &Invocation
+}
+```
 
-  # Task authorization.
-  auth    &Authorization
+### 5.1.1 Task
 
-  meta    {String : any}
+The `run` field MUST contain a link to the [Task] to be run.
 
-  prf     [&UCAN]
-}
+### 5.1.2 Metadata
+
+The OPTIONAL `meta` field MAY be used to include human-readable descriptions, tags, execution hints, resource limits, and so on. If present, the `meta` field MUST contain a map with string keys. The contents of the map are left undefined to encourage extensible use.
+
+If `meta` field is not present, it is implicitly a `unit` represented as an empty map.
 
-type SemVer string
+### 5.1.3 Proofs
+
+The `prf` field MUST contain links to any UCANs that provide the authority to perform this task. All of the outermost proofs MUST have `aud` field set to the [Executor]'s DID. All of the outmost proofs MUST have `iss` field set to the [Invoker]'s DID.
+
+### 5.1.4 Cause
+
+[Task]s MAY be invoked as an effect caused by a prior [Invocation]. Such [Invocation]s SHOULD have a `cause` field set to the [Receipt] link of the [Invocation] that caused it. The linked [Receipt] MUST have an `Effect` (the `fx` field) containing invoked [Task] in the `run` field.
+
+## 5.2 (Signed) Invocation
+
+An `Invocation` is a signed `Context`.
+
+```ipldsch
+type Invocation struct {
+  ctx     Context
+  sig     Varsig
+}
 ```
 
-## 5.2 Fields
+### 5.2.1 Context
 
-### 5.2.1 UCAN Task Version
+The `Context` containing the `Task` and any configuration.
 
-The `v` field MUST contain the SemVer-formatted version of the UCAN Invocation Specification that this struct conforms to.
+### 5.2.2 Signature
 
-### 5.2.2 Task
+The `sig` field MUST contain a [varsig] of the `Context`, signed by the issuer of the proofs.
 
-The `run` field MUST contain a link to the [Task] to be run.
+## 5.3 Capsule
 
-### 5.2.3 Cause
+An invocation capsule associates the `Invocation` with a versioned schema. This MAY be omitted in contexts where the schema and version are clear from context (for example, when nested in another structure that defines the version)
 
-[Task]s MAY be invoked as an effect caused by a prior [Invocation]. Such [Invocation]s SHOULD have a `cause` field set to the [Receipt] link of the [Invocation] that caused it. The linked [Receipt] MUST have an `Effect` (the `fx` field) containing invoked [Task] in the `run` field.
+```ipldsch
+type InvocationCapsule struct {
+  inv     Invocation (rename "ucan/invoke@0.2.0")
```

Comment:
@expede So, I get that rename refers to the serialized version of this structure. In terms of implementation, since the version is dynamic, it seems like an implementation would use a `vsn` field here anyway as the "key" in a struct (thinking like in Rust, since we wouldn't want to represent this as a hashmap), right?

Reply:
Of note, the same question refers to the receipt / rename below.

Reply:
I think in rust this would translate to an enum with option per supported version

Reply:
yes in terms of your question. 

```
Breaking this into a capsule means that you can omit it where it's clear from context
```

I guess, though, you're still having to know which shape of an invocation and inner context represented which version. That's essentially more like the enum per version type that @Gozala alluded to, which seems unnecessary?

```
There's less parsing required to get to the discriminant if it's on the outside instead of embedded in any arbitrary order of an unboundedly large struct (which is a minor performance gain, but zero)
```

Eh. Again, many API version fields live inside protos for that API, right?